webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala

```diff
@@ -0,0 +1,1424 @@
+
+            //_ = log.debug("userDataByIRIGetV1 - sparqlQueryString: {}", sparqlQueryString)
+
+            userDataQueryResponse <- (storeManager ? SparqlSelectRequest(sparqlQueryString)).mapTo[SparqlSelectResponse]
+
+            groupedUserData: Map[String, Seq[String]] = userDataQueryResponse.results.bindings.groupBy(_.rowMap("p")).map {
+                case (predicate, rows) => predicate -> rows.map(_.rowMap("o"))
+            }
+
+            /* the projects the user is member of */
+            projectIris: Seq[IRI] = groupedUserData.get(OntologyConstants.KnoraBase.IsInProjectAdminGroup) match {
+                case Some(projects) => projects
+                case None => Seq.empty[IRI]
+            }
+
+            maybeProjectFutures: Seq[Future[Option[ProjectADM]]] = projectIris.map {
+                projectIri => (responderManager ? ProjectGetADM(maybeIri = Some(projectIri), maybeShortcode = None, maybeShortname = None, requestingUser = KnoraSystemInstances.Users.SystemUser)).mapTo[Option[ProjectADM]]
+            }
+            maybeProjects: Seq[Option[ProjectADM]] <- Future.sequence(maybeProjectFutures)
+            projects: Seq[ProjectADM] = maybeProjects.flatten
+
+            // _ = log.debug("userProjectAdminMembershipsGetRequestV1 - userIri: {}, projectIris: {}", userIri, projectIris)
+        } yield projects
+    }
+
+    /**
+      * Returns the user's project admin group memberships, where the result contains the IRIs of the projects the user
+      * is a member of the project admin group.
+      *
+      * @param userIri       the user's IRI.
+      * @param requestingUser the requesting user.
+      * @param apiRequestID  the unique api request ID.
+      * @return a [[UserProjectMembershipsGetResponseV1]].
+      */
+    def userProjectAdminMembershipsGetRequestADM(userIri: IRI, requestingUser: UserADM, apiRequestID: UUID): Future[UserProjectAdminMembershipsGetResponseADM] = {
+
+        // ToDo: which user is allowed to do this operation?
+        // ToDo: check permissions
+
+        for {
+            userExists <- userExists(userIri)
+            _ = if (!userExists) {
+                throw BadRequestException(s"User $userIri does not exist.")
+            }
+
+            projects: Seq[ProjectADM] <- userProjectAdminMembershipsGetADM(userIri = userIri, requestingUser = KnoraSystemInstances.Users.SystemUser, apiRequestID = apiRequestID)
+        } yield UserProjectAdminMembershipsGetResponseADM(projects = projects)
+    }
+
+    /**
+      * Adds a user to the project admin group of a project.
+      *
+      * @param userIri       the user's IRI.
+      * @param projectIri    the project's IRI.
+      * @param requestingUser the requesting user.
+      * @param apiRequestID  the unique api request ID.
+      * @return
+      */
+    def userProjectAdminMembershipAddRequestV1(userIri: IRI, projectIri: IRI, requestingUser: UserADM, apiRequestID: UUID): Future[UserOperationResponseADM] = {
```

Comment:
Why isn't this `userProjectAdminMembershipAddRequestADM`?

Reply:
forgot to rename it.

Reply:
done